src/components/Reviews/ReviewPage/ReviewPage.js

```diff
@@ -63,6 +50,161 @@ const ReviewPage = (props) => {
+
+            Object.entries({ ...fields, file }).forEach(([key, value]) => {
+                formData.append(key, value);
+            });
+
+            const upload = await fetch(url, {
+                method: 'POST',
+                body: formData,
+            });
+
+            if (upload.ok) {
+                console.log('Uploaded successfully!');
+            } else {
+                console.error('Upload failed.');
+            }
+        };
+
+
+        for (let i = 0; i < files.length; i++) {
+            await uploadPhoto(fileRecords[i])
+        }
+
+
+        fetch(`${rootDomain}/review/add`, {
+            method: 'POST',
+            body: JSON.stringify(body),
+            headers: {
+                'Content-Type': 'application/json',
+                'X-CSRF-TOKEN': Cookies.get('csrf_access_token'),
+            },
+        }).then(response => {
+            if (response.ok) {
+                sendEvent('review_submit', {
+                    'site_id': body.beach_id,
+                });
+                Router.push(`/Beach/${body['beach_id']}`)
+            } else {
+                response.json().then(({ msg }) => toaster.danger(msg));
+            }
+            return response.json()
+        })
+    }
+
+    const DropZoneArea = () => {
+
+        const { acceptedFiles, getRootProps, getInputProps } = useDropzone({ accept: 'image/*' });
+
+        return (
+            <div>
+                {acceptedFiles.length < 2 &&
+                    <div {...getRootProps({ className: 'dropzone' })}>
+                        <input {...getInputProps()} />
+                        <div className={styles.dropzone}>
+                            <div className={styles.fileupload}>
+                                <div className={styles.plusicon}>
+                                    +
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                }
+                <FileSubmit actualFile={acceptedFiles}></FileSubmit>
+            </div>
+        )
+    }
+
+
+
+
+    function FileSubmit({ actualFile }) {
+
+        let f = null;
+
+        if (actualFile) {
+            actualFile.map(file => {
+                f = file;
+            })
+        }
+
+        React.useEffect(() => {
+
+            if (f) {
+                const rand = (Math.random()).toString().substring(3, 10)
+                let testid = rand + f.path
+                const myNewFile = new File([f], testid, { type: f.type });
+                const objUrl = URL.createObjectURL(myNewFile)
+                let newFileRecord = {
+                    file: myNewFile,
+                    url: objUrl
+                }
+                let fileRecordsCopy = fileRecords
+                fileRecordsCopy.push(newFileRecord)
+                setFileRecords([...fileRecordsCopy])
+                
+
+            }
+
+        }, [f]);
+
+        return (
```

Comment:
It's definitely a little hacky but that function is called twice when the photo is uploaded so that patterns prevents the same picture being updated to the file records twice by only being called when f changes. Probably would've been better to lay out the architecture beforehand but I'm still a little off on the re-rendering and when that happens.